### PR TITLE
Updates for v2.0 Node 20

### PR DESCRIPTION
**Pull request by nonword**
Updates so we can get this thing on Node20:
 - Use v2 of nypl-data-api-client
 - Use v3 AWS SDK
 - Many small dependency updates for a clean `npm audit`
 - Linting

Also, dropping support for createStream and deleteStream, since they are dangerous actions and never used.

Also, changing default branch from `master` to `main` (after merging).

This is a major 2.0.0 release and will be published to NPMJS after merging.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,30 +14,36 @@ npm i @nypl/nypl-streams-client --save
...
 ### Example 1: Writing data to a stream
 
 To write a single record to a stream (encoded to "MyStream" schema):
+
 ```js
-streamsClient.write('MyStream', { id: 'id1', field1: 1, field2: 2 }).then((resp) => {
-  console.log('Finished writing to stream ' + resp.Records.length)
-}).catch((e) => console.error('Error writing to stream: ', e))
+streamsClient.write('MyStream', { id: 'id1', field1: 1, field2: 2 })
+  .then((resp) => {
```
Comment (charmingduchess) [inline README.md:28]: Can you convert this to try/catch for consistency?

File: README.md [hunk 2]
```diff
@@ -14,30 +14,36 @@ npm i @nypl/nypl-streams-client --save
...
+const records = [ { id: 'id1', field1: 1, field2: 2 }, { id: 'id2', field1: 1 }, ... ] // Array of any length
+const options = {
   recordsPerSecond: 500 // This is the default and well below the 1000/s AWS constraint
 }
-streamsClient.write('MyStream', records, options).then((resp) => {
-  console.log('Finished writing to stream ' + resp.Records.length)
-  console.log(`Failed to write: ${resp.FailedRecordCount} record(s)`)
-}).catch((e) => console.error('Error writing to stream: ', e))
+streamsClient.write('MyStream', records, options)
+  .then((resp) => {
```
Comment (charmingduchess) [inline README.md:42]: ditto for try/catch here!

**Review by charmingduchess — COMMENTED** [review]
Looks good! Most of the PR looks like it's using try/catch, so let's standardize and replace the remaining `.then().catch()` implementations.

File: README.md (hunk 1 above)
Comment (nonword) [reply README.md]: Updated

**Review by charmingduchess — COMMENTED** [review]
Looking through this again, I'm seeing a bunch of nested promise callbacks (like  [here](https://github.com/NYPL/node-nypl-streams-client/pull/30/files#diff-ca407d959d33ce92a7f7efee354a5ea5e6e61fcd797d32c04205e429a074ed54R126)). I'd like to see them converted into try/catch situations, but I don't want to bog down the completion of the work. If you'd like to update them, please do.

**Comment by nonword** [comment]
Yes this was primarily a platform/dependency vulnerability update - less concerned with style. There's a bit I'd change in a proper refactor. So going to merge as is and we can come back to use of async/await?

**Pull request merged**